### DISCUSSION TITLE
Factors limiting speed - only about 30-40 MBs no matter what.

### DISCUSSION THREAD

**Discussion by Frankieeeeeeeeeeeee** [outdoorbits/little-backup-box#229] [Q&A]
Hi guys, 
so I have purchased RPi 4 8GB variant and make LBB work. It works well, but I am quite surprised about copy speed. As per the logs (and timing also confirms) it is only about 30-40 MB/s.
I tried it on larger files (see logs) as I thought the speed could be caused by transferring many little ones. But it is still the same.
I am using SanDisk Extreme 128 GB U3 A2 card in the LBB and the same one in USB 3.1 card reader (I tried two different ones). When I plug in the card reader directly to PC, I get about 90 MB/s read speed.

Also I have tried copy from USB mounted SSD to USB mounted SSD (in powered USB hub) - again big files. I got only about 40 MB/s on avarage on that as well :/. When both SSDs are plugged into laptop, I get around 330 MB/s copy speed. Of course I don't expect that much on RPi, but current reality is way below expectation...

Are there any common mistakes/places to check what I could do? I am thinking whether or not to keep the RPi (I still have a week to return it for full refund...)

Thanks!
[little-backup-box (3).log](https://github.com/outdoorbits/little-backup-box/files/12373951/little-backup-box.3.log)
![lbb_stats](https://github.com/outdoorbits/little-backup-box/assets/141441147/1a98855f-265b-45c4-ab9e-ebc5e0658047)
[little-backup-box (2).log](https://github.com/outdoorbits/little-backup-box/files/12373952/little-backup-box.2.log)

Logs: 
2023/08/11 14:27:25 [11090] sent 29.74G bytes  received 5.18K bytes  39.69M bytes/sec
2023/08/11 14:27:25 [11090] Number of created files: 289 (reg: 263, dir: 26)

**Comment by outdoorbits** [comment]
In another discussion there was the idea to use the whole file option (rsync) for local syncs. The upcoming rewritten version will include this. But this will take about two more weeks.
I'm still traveling, so I can't test it. But maybe you would test performance difference using this option? I'm very interested!

**Comment by rkosick** [comment]
What file system are you using? With my raspberry pi zero 2, the speed decreases using NTFS formatted SSDs by quite a bit.
@outdoorbits You mention in the news that it is rewritten to python now. I can't see details in the installation instruction. Is it available through the development branch or already included completely?

**Comment by outdoorbits** [comment]
The rewritten version is installable by a special command. You find it directly below the standard installation command on GitHub. For local backups from USB it should be really faster. I'm using exfat but more for general compatibility reasons.

**Comment by outdoorbits** [comment]
Just install the devolvement branch by this command:
curl -sSL https://raw.githubusercontent.com/outdoorbits/little-backup-box/development/install-little-backup-box.sh | bash -s -- development 2> install-error.log

I'm looking forward for feedback!

**Reply by rkosick** [reply]
Yes this is what I meant. I'll give it a try.

**Reply by outdoorbits** [reply]
Backup USB to USB I just got around 120 MB/s!
